adoc/chapters/device_compiler.adoc

```diff
@@ -216,132 +217,25 @@ The restriction waiver in <<discarded-statement>> or
 <<device-function>>.
 ====
 
-[[subsec:scalartypes]]
```

Comment:
Note that the latest version of the "main" branch has text that refers to this section identifier.  (This was part of the recent `vec` clarification.)  You will need to reword that new text somehow to avoid the reference.